src/main/ui/src/components/Login/Login.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+import basestyle from "./Base.module.css";
+import loginstyle from "./Login.module.css";
+import axios from "axios";
+import { useNavigate, NavLink } from "react-router-dom";
+const Login = ({ setUserState }) => {
+  const navigate = useNavigate();
+  const [formErrors, setFormErrors] = useState({});
+  const [isSubmit, setIsSubmit] = useState(false);
+  const [user, setUserDetails] = useState({
+    email: "",
+    password: "",
+  });
+
+  const changeHandler = (e) => {
+    const { name, value } = e.target;
+    setUserDetails({
+      ...user,
+      [name]: value,
+    });
+  };
+  const validateForm = (values) => {
+    const error = {};
+    const regex = /^[^\s+@]+@[^\s@]+\.[^\s@]{2,}$/i;
+      if (!values.fname) {
+          error.fname = "First Name is required";
+        }
+    if (!values.email) {
+      error.email = "Email is required";
+    } else if (!regex.test(values.email)) {
+      error.email = "Please enter a valid email address";
+    }
+    if (!values.password) {
+      error.password = "Password is required";
+    }
+    return error;
+  };
+
+  const loginHandler = (e) => {
+    e.preventDefault();
+    setFormErrors(validateForm(user));
+    setIsSubmit(true);
+    // if (!formErrors) {
+
+    // }
+  };
+
+  useEffect(() => {
+    if (Object.keys(formErrors).length === 0 && isSubmit) {
+      console.log(user);
+      axios.post("http://localhost:3000/login", user).then((res) => {
+        alert(res.data.message);
+        setUserState(res.data.user);
+        navigate("/", { replace: true });
+      });
+    }
+  }, [formErrors]);
```

Comment:
getting error in VSCode

`const formErrors: {}
React Hook useEffect has missing dependencies: 'isSubmit', 'navigate', 'setUserState', and 'user'. Either include them or remove the dependency array. If 'setUserState' changes too often, find the parent component that defines it and wrap that definition in useCallback`